endpoints/openrtb2/amp_auction.go

```diff
@@ -230,6 +230,22 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 		return
 	}
 
+	// Populate any "missing" OpenRTB fields with info from other sources, (e.g. HTTP request headers).
+	deps.setFieldsImplicitly(r, reqWrapper, account)
+
+	hasStoredResponses := len(storedAuctionResponses) > 0
+	errs := deps.validateRequest(reqWrapper, true, hasStoredResponses, storedBidResponses, false)
+	errL = append(errL, errs...)
+	ao.Errors = append(ao.Errors, errs...)
+	if errortypes.ContainsFatalError(errs) {
+		w.WriteHeader(http.StatusBadRequest)
+		for _, err := range errortypes.FatalOnly(errs) {
+			w.Write([]byte(fmt.Sprintf("Invalid request: %s\n", err.Error())))
+		}
+		labels.RequestStatus = metrics.RequestStatusBadInput
+		return
+	}
+
```

Comment:
@SyntaxNode moving this out of `parseAmpRequest` which happens upstream, appears safe to me. Now we are initializing targeting and caching, reading user syncs/cookies and retrieving the account prior to setting fields implicitly and validating the request but that seems ok. Does this look safe to you too?

I also confirmed that moving this logic has no implication on modularity infrastructure as this logic still happens after the entrypoint stage (injection point 1) and before the processed auction stage (injection point 3). The raw auction stage (injection point 2) is not relevant to AMP.

Reply:
Test for error scenario is already covered in topics PR, let me know if you want it to be moved in this PR.